plugins/miopen_legacy_plugin/engines/solvers/miopen_batchnorm_solver.cpp

```
@@ -2,21 +2,187 @@
+
+    //todo, below are optional, need to check if they exist and only set then
+    create_tensor_and_device_buffer_pair(
+        *tensor_map.at(attributes.mean().value()), device_buffers, num_device_buffers, estMeanDesc);
+
+    create_tensor_and_device_buffer_pair(*tensor_map.at(attributes.inv_variance().value()),
+                                         device_buffers,
+                                         num_device_buffers,
+                                         estVarianceDesc);
+
+    auto miopen_status
+        = miopenBatchNormalizationForwardInference_V2(handle.miopen_handle,
+                                                      miopen_batchnorm_mode,
+                                                      &alpha,
+                                                      &beta,
+                                                      xDesc.tensor_desc,
+                                                      xDesc.device_buffer.ptr,
+                                                      yDesc.tensor_desc,
+                                                      yDesc.device_buffer.ptr,
+                                                      scaleDesc.tensor_desc,
+                                                      biasDesc.tensor_desc,
+                                                      estMeanDesc.tensor_desc,
+                                                      estVarianceDesc.tensor_desc,
+                                                      scaleDesc.device_buffer.ptr,
+                                                      biasDesc.device_buffer.ptr,
+                                                      estMeanDesc.device_buffer.ptr,
+                                                      estVarianceDesc.device_buffer.ptr,
+                                                      epsilon);
+
+    HIPDNN_LOG_INFO("MIOpen batchnorm forward inference status: {}",
+                    miopenGetErrorString(miopen_status));
```

Maybe throw an error here if status isn't success? Or could generally wrap miopen API calls in some sort of `MIOPEN_CHECK` macro:
```cpp
#define MIOPEN_CHECK(status)                                                             \
    do                                                                                   \
    {                                                                                    \
        if(status != miopenStatusSuccess)                                                \
        {                                                                                \
            std::cerr << "MIOpen Error: " << miopenGetErrorString(status) << " in file " \
                      << __FILE__ << " at line " << __LINE__ << std::endl;               \
            exit(EXIT_FAILURE);                                                          \
        }                                                                                \
    } while(0)
```

---

agreed, I will do that.  Theres a few things in the solver I still need to address from a code standpoint.  The struct I used to combine tensor/device needs to be extracted and the mapping function is likely a useful extraction as well.  I assume most miopen solvers will need to map tensor to the raw buffer.

---

added two miopen checks.  1 that throws and another that doesnt